src/NewTokenStandard.ts

```diff
@@ -250,12 +252,63 @@ class FungibleToken extends TokenContract {
   }
 
   @method.returns(AccountUpdate)
-  async mint(
+  async mintWithProof(
     recipient: PublicKey,
     amount: UInt64,
     proof: SideloadedProof,
     vk: VerificationKey, // provide the full verification key since only the hash is stored.
     vKeyMap: VKeyMerkleMap
+  ): Promise<AccountUpdate> {
+    const packedDynamicProofConfigs =
+      this.packedDynamicProofConfigs.getAndRequireEquals();
+    const mintDynamicProofConfig = MintDynamicProofConfig.unpack(
+      packedDynamicProofConfigs
+    );
+
+    await this.verifySideLoadedProof(
+      proof,
+      vk,
+      recipient,
+      mintDynamicProofConfig,
+      vKeyMap,
+      OperationKeys.Mint
+    );
+
+    return await this.internalMint(recipient, amount);
+  }
+
+  /**
+   * Mints tokens to a recipient without requiring side-loaded proof verification.
+   * This function can only be used when dynamic proof verification is disabled in the mint configuration.
+   *
+   * @param recipient - The public key of the account to receive the minted tokens
+   * @param amount - The amount of tokens to mint
+   * @returns The account update for the mint operation
+   * @throws {Error} If dynamic proof verification is enabled in the mint configuration
+   * @throws {Error} If the recipient is the circulation account
```

Comment:
I like this pattern you have here of adding errors paths that will throw to the doc comments! Not a specific ask, but maybe we can incorporate this pattern elsewhere in our code going forward.